app/src/main/kotlin/com/wire/android/ui/home/conversationslist/ConversationRoute.kt

```diff
@@ -36,23 +35,23 @@ fun ConversationRoute(viewModel: ConversationViewModel = ConversationViewModel()
                         AllConversationScreen(
                             newActivities = newActivities,
                             conversations = conversations
-                        ) { }
+                        ) { conversationId -> conversationListViewModel.openConversation(conversationId) }
```

Comment:
Maybe it would be cleaner to simply inject the viewModel on each Conversations Screen, and let them invoke there the logic to navigate to the appropriate Detail Conversation with the right conversation id.

Reply:
I do not think it makes much sense as we are sharing a ViewModel anyway between those 3 Composable elements and just propagate the state changes from the viewmodel to that screen, so in my opinion it makes more sense to push the click events upto ConversationRoute and let there handle it.